coalib/bears/Bear.py

```diff
@@ -324,6 +335,32 @@ def log_message(self, log_message, timestamp=None, **kwargs):
     def run(self, *args, dependency_results=None, **kwargs):
         raise NotImplementedError
 
+    def _dump_bear_profile_data(self, profiler):
+        filename = '{}_{}.prof'.format(self.section.name, self.name)
+        if self.profile is True:
+            profiler.dump_stats(filename)
```

Comment:
dont have two invocations of `profiler.dump_stats` in different branches unless they are doing different things.

your logic should be

1. determine `filename`
2. invoke `dump_stats`

dont mix them together.

Reply:
@jayvdb  didn't get, don't know how am I going to tackle this condition where all the things get satisfied because if `self.profile` value is type `bool` so `os.path.join` will throw error and can create further problem which I have to handle just because to make `one invocations of profiler.dump_stats`

>determine filename

This is not the only case if self.profile will be `True` it will dump the profiled files in the current working directory

Reply:
What @jayvdb means is that you have duplicated the logic regarding the invocation of `profiler.dump_stats`: In any case, you are dumping things here. So to avoid redundancy, do it like suggested:

> 1. determine filename
> 2. invoke dump_stats

means

```python
if self.profile is True:
    path = ...
else:
    path = ...

profiler.dump_stats(path)
```

because the difference is the parameter you pass to `dump_stats`, not the call.

I believe we already discussed that somewhen ^^

Reply:
That was different code but same code duplication was there :+1: